### DISCUSSION TITLE
No liability to discuss for aiken-kan @KtorZ don't discuss full and ban me why

### DISCUSSION THREAD

**Discussion by avi69night** [cardano-foundation/developer-portal#1338] [General]
i was experiment in aiken 
[[KtorZ](https://github.com/KtorZ)](https://github.com/aiken-lang/aiken/discussions/1039)
he said it wasn't aiken code yes i know just ban me without listing full discuss

The main aiken code is
.....
fn select_choice(choices: List<String>, probabilities: List<Float>) -> String {
    // Ensure that probabilities sum up to 1.0
    let sum_probs = List.fold(probabilities, 0.0, fn prob, acc -> acc + prob end)

    assert(sum_probs == 1.0, "Probabilities must sum to 1.0")

    let rand_value = Plutus.rand_float()  // Assuming a random float function exists in Plutus/Aiken
    let cumulative_prob = 0.0

    let choice = List.fold(
        List.zip(choices, probabilities),
        None,
        fn (choice_prob, acc_prob) ->
            let (choice, prob) = choice_prob
            let updated_prob = acc_prob + prob

            if rand_value < updated_prob {
                Some(choice)  // Select this choice
            } else {
                acc_prob
            }
        end
    )

    match choice {
        Some(selected_choice) -> selected_choice
        None -> List.head(choices)  // Return the first option if no match
    }
}
...

**Comment by Plutusart2024** [comment]
this code have issues but it is aiken code


why not people get to chance for dis cuss

**Comment by danny33Boy** [comment]
The issues of this code is 

Randomness (Plutus.rand_float()):

List.fold 
acc_prob Handling:

**Comment by rdlrt** [comment]
Dev Portal is not the place for this discussion, please use corresponding channels instead (also, keep to single profile...you're [atleast] coming across as spamming with different profiles, basic common sense and github decorum is certainly missing)

**Comment by rphair** [comment]
This posting is off topic for this repository, so one of my associates or I will probably delete it soon.  Please raise any enquiries here instead: Cardano Foundation Engineering & Development - invitation: https://discord.gg/XVcn9CW3
